modules/integration/src/test/scala/scala/cli/integration/BspTestDefinitions.scala

```diff
@@ -149,6 +149,39 @@ abstract class BspTestDefinitions(val scalaVersionOpt: Option[String])
     }
   }
 
+  val importPprintOnlyProject = TestInputs(
+    Seq(
+      os.rel / "simple.sc" -> s"import $$ivy.`com.lihaoyi::pprint:${Constants.pprintVersion}`"
+    )
+  )
+
+  test("setup-ide should succeed for valid dependencies") {
+    importPprintOnlyProject.fromRoot { root =>
+      val p = os.proc(
+        TestUtil.cli,
+        "setup-ide",
+        ".",
+        extraOptions,
+        "--dependency",
+        s"org.scalameta::munit:${Constants.munitVersion}"
+      ).call(cwd = root)
+    }
+  }
+
+  test("setup-ide should fail for invalid dependencies") {
+    importPprintOnlyProject.fromRoot { root =>
+      val p = os.proc(
+        TestUtil.cli,
+        "setup-ide",
+        ".",
+        extraOptions,
+        "--dependency",
+        "org.scalameta::munit:0.7.119999"
+      ).call(cwd = root, check = false)
+      expect(p.exitCode == 1)
```

Comment:
By passing `stderr = os.Pipe` to `call`, we can get the errors printed by scala-cli, and ensure that the error originate from a not found dependency.